controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go

```diff
@@ -238,16 +239,21 @@ func (r *KubeadmControlPlaneReconciler) reconcile(ctx context.Context, cluster *
 			r.recorder.Eventf(kcp, corev1.EventTypeWarning, "FailedInitialization", "Failed to initialize the control plane: %v", err)
 			return ctrl.Result{}, err
 		}
+		if numMachines+1 < desiredReplicas {
+			return ctrl.Result{Requeue: true}, nil
+		}
 	// scaling up
 	case numMachines < desiredReplicas && numMachines > 0:
```

Comment:
Did you try collapsing these cases? I almost want to do that, but I'm not sure if it'd be better or not. This might be clearer to read with only two paths, scale up and scale down. But honestly, this is kind of a nit so feel free to ignore

Reply:
I thought about it, but decided to put off refactoring for later, in order to keep the changes easier to understand. I can follow up with another commit in this PR. :smile: